src/dataset-builder/CohortEditor.test.ts

```diff
@@ -479,7 +479,7 @@ describe('CohortEditor', () => {
     await user.click(screen.getByText('Save cohort'));
     // Assert
     // Don't compare name since it's generated.
-    const { name: _unused, ...expectedCriteriaGroup } = newCriteriaGroup();
+    const { name: _unused, ...expectedCriteriaGroup } = newCriteriaGroup(originalCohort.criteriaGroups.length + 1);
     expect(updateCohorts.mock.calls[0][0]([])).toMatchObject([
       { ...originalCohort, criteriaGroups: [expectedCriteriaGroup] },
```

Comment:
I think an added test to show the new expected functionality would be good. I'd be happy to pair if you want